hs_tracking/views.py

```diff
@@ -4,9 +4,39 @@
 from django.views.generic import TemplateView
 from django.utils.decorators import method_decorator
 from django.contrib.auth.decorators import user_passes_test
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseRedirect
 
 from . import models as hs_tracking
+from .models import Session, Variable
+from .utils import get_std_log_fields
+
+
+class AppLaunch(TemplateView):
+
+    def get(self, request, **kwargs):
+
+        # get the url or hydroshare.org if one is not provided.
+        url = request.GET.get('url', 'http://www.hydroshare.org')
+
+        # log app launch details if user is logged in
+        if hasattr(request, 'user'):
```

Comment:
I thing this condition will always be true. May be the following is what you need:

```if request.user.is_authenticated():```